src/main/java/org/ibp/api/java/impl/middleware/ontology/brapi/VariableServiceBrapiImpl.java

```diff
@@ -42,12 +47,37 @@ public long countObservationVariables(final VariableSearchRequestDTO requestDTO)
 	public VariableUpdateResponse updateObservationVariable(final String observationVariableDbId, final VariableDTO variable) {
 		final VariableUpdateResponse variableUpdateResponse = new VariableUpdateResponse();
 		try {
-			this.variableUpdateValidator.validate(observationVariableDbId, variable);
+			this.variableDtoValidator.validateForUpdate(observationVariableDbId, variable);
 			variableUpdateResponse.setEntityObject(this.middlewareVariableServiceBrapi.updateObservationVariable(variable));
 		} catch (final ApiRequestValidationException e) {
 			variableUpdateResponse.setEntityObject(variable);
 			variableUpdateResponse.setErrors(e.getErrors());
 		}
 		return variableUpdateResponse;
 	}
+
+	@Override
+	public VariableImportResponse createObservationVariables(final String crop, final List<VariableDTO> variableDTOList) {
+
+		final VariableImportResponse response = new VariableImportResponse();
+		response.setImportListSize(variableDTOList.size());
+		response.setCreatedSize(0);
+
+		final BindingResult bindingResult = this.variableDtoValidator.pruneVariablesInvalidForImport(variableDTOList);
+		if (bindingResult.hasErrors()) {
+			response.setErrors(bindingResult.getAllErrors());
+		}
+
+		final List<VariableDTO> variables = this.middlewareVariableServiceBrapi.createObservationVariables(variableDTOList);
+		if (CollectionUtils.isNotEmpty(variables)) {
+			final VariableSearchRequestDTO variableSearchRequestDTO = new VariableSearchRequestDTO();
+			// Fetch the saved observation variables from the database to populate the fields.
+			variableSearchRequestDTO.setObservationVariableDbIds(variables.stream().map(VariableDTO::getObservationVariableDbId).collect(
+				Collectors.toList()));
+			final List<VariableDTO> createdVariables = this.getObservationVariables(crop, variableSearchRequestDTO, null);
```

Comment:
@abatac is there any way to avoid querying variables if the createObservationVariables is already returning the list of VariableDTO? Maybe completing attributes that are missing in middleware will help us to avoid this extra call to get

Reply:
Hi Clary, I've fixed this. I moved the logic to retrieve the variableDTOs to make sure the fields are populated directly from middleware VariableServiceBrapiImpl#createObservationVariables.